spring-integration-core/src/main/java/org/springframework/integration/channel/MessagePublishingErrorHandler.java

```diff
@@ -70,9 +73,24 @@ public void setDefaultErrorChannel(MessageChannel defaultErrorChannel) {
 	 * @since 4.3
 	 */
 	public MessageChannel getDefaultErrorChannel() {
+		String defaultErrorChannelName = this.defaultErrorChannelName;
+		if (defaultErrorChannelName != null) {
+			if (this.channelResolver != null) {
```

Comment:
In other places (e.g. AMPH), we do this in a synchronized block - did you make a conscious decision not to here, because a null channel is common? That said, it's idempotent so maybe we should remove the synchronization in other places?